### DISCUSSION TITLE
Auth (GRAPH) deprecation (june/july 2022) and revokation

### DISCUSSION THREAD

**Discussion by danielfdickinson** [abraunegg/onedrive#1697] [Q&A] [answered]
I noticed the the GRAPH API this app is using is slated for removal in June/July of 2022. Is there a plan to update to it's replacement (which will have the benefit of more fine-grained permissions granting) before then?

Also, if I want to revoke the token that grants the app permissions to my onedrive, how do I do that?

**Comment by abraunegg** [comment, marked as answer]
@danielfdickinson 
> I noticed the the GRAPH API this app is using is slated for removal in June/July of 2022. Is there a plan to update to it's replacement (which will have the benefit of more fine-grained permissions granting) before then?

There is no plan, as this client is not impacted. This client does not use `graph.windows.net` which is the service that is shutting down.

What is being shutdown is this: Azure AD Graph API

This application uses Microsoft Graph API which is not being shutdown, and this is what this client uses.

> Also, if I want to revoke the token that grants the app permissions to my onedrive, how do I do that?

All you need to do is issue: `onedrive --logout` and this revokes your token.

**Reply by danielfdickinson** [reply]
Thanks. I misunderstood about the GRAPH API and didn't connect ``--logout`` without ``revoke``.
